app/services/container_topology_service.rb

```diff
@@ -80,10 +80,14 @@ def entity_status(entity, kind)
     when 'VM', 'Host' then entity.power_state.capitalize
     when 'Node'
       condition = entity.container_conditions.first
-      if condition.name == 'Ready' && condition.status == 'True'
-        'Ready'
+      if condition.nil?
+        'Unknown'
       else
-        'NotReady'
+        if condition.name == 'Ready' && condition.status == 'True'
+          'Ready'
+        else
+          'NotReady'
+        end
```

Comment:
> This looks much better than just determining status from the first condition. If this needs to be merged urgently, I can do so if the approach suggested by @simon3z is implemented in a follow-up PR.

I also agree with the suggested approach and I think it should be done in another PR, however I also think that this PR fixes a bug, so tests should be updated accordingly as part of the PR.